internal/store/store.go

```diff
@@ -269,8 +269,22 @@ func WithContext(ctx context.Context) Option {
 	}
 }
 
+// Appends user agent given a build context for telemetry.
+func appendUserAgent(buildContext string) {
+	if !strings.Contains(config.UserAgent, buildContext) {
+		config.UserAgent = fmt.Sprintf("%s-%s", config.UserAgent, buildContext)
+	}
+}
+
 // setAtlasClient sets the internal client to use an Atlas client and methods.
 func (s *Store) setAtlasClient(client *http.Client) error {
+	buildContext, buildContextPresent := os.LookupEnv("BUILD_CONTEXT")
+
+	if buildContextPresent {
+		appendUserAgent(buildContext)
+	}
+
+	fmt.Println(config.UserAgent)
```

Comment:
From what I can see, the UserAgent is set from 
https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/oauth/oauth.go#L70C11-L70C11

I wonder would it be better to pass the value from there, since it should only have to be called once when authenticating the CLI

Reply:
I agree it seems like the better place to put it, updated